### PR TITLE
transformer c files

### PR DESCRIPTION
**Pull request by xingfeng01**
### PR types
New features 

### PR changes
OPs

### Describe
This PR adds fused transformer related files defining c interface including class, function etc..

### CODE REVIEW

File: paddle/fluid/operators/fused/fused_transformer_op.cc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+#include <string>
+
+namespace paddle {
+namespace operators {
+
+// constructor and init
+template <typename T>
+FusedTransformerEncoderLayer<T>::FusedTransformerEncoderLayer(
+    int batch_size_, int max_seq_len_, int dim_embed_, int dim_feedforward_,
+    int num_head_, T dropout_, T act_dropout_, T attn_dropout_,
```
Comment (zlsh80826) [inline paddle/fluid/operators/fused/fused_transformer_op.cc]: dropout type can be fix to float

File: paddle/fluid/operators/fused/fused_transformer_op.cc [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+template <typename T>
+void FusedFFN<T>::ComputeBackward() {}
+
+// init
+template <typename T>
+FusedMHA<T>::FusedMHA() {}
+
+// compute forward
+template <typename T>
+void FusedMHA<T>::ComputeForward() {}
```
Comment (zlsh80826) [inline paddle/fluid/operators/fused/fused_transformer_op.cc]: FusedMHA<T>::ComputeForward(int batch size, int max_seq_len, int num_head, int head_size, float dropout, bool is_test, uint64_t seed, const T* x, const int32_t* seqlen, const int32_t* cu_seqlen, T* output, T* softmax_mask) {}

File: paddle/fluid/operators/fused/fused_transformer_op.cc [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+// compute backward
+template <typename T>
+void FusedMHA<T>::ComputeBackward() {}
```
Comment (zlsh80826) [inline paddle/fluid/operators/fused/fused_transformer_op.cc]: FusedMHA::ComputeBackward(int batch size, int max_seq_len, int num_head, int head_size, float dropout, const T* x, const int32_t* seqlen, const int32_t* cu_seqlen, const T* grad_output, T* softmax_mask, T* grad_x) {}

File: paddle/fluid/operators/fused/fused_transformer_op.cc (hunk 1 above)
Comment (xingfeng01) [reply paddle/fluid/operators/fused/fused_transformer_op.cc]: updated

File: paddle/fluid/operators/fused/fused_transformer_op.cc (hunk 2 above)
Comment (xingfeng01) [reply paddle/fluid/operators/fused/fused_transformer_op.cc]: updated

Comment (xingfeng01) [reply paddle/fluid/operators/fused/fused_transformer_op.cc]: updated

File: paddle/fluid/operators/fused/fused_transformer_op.cc (hunk 3 above)
Comment (xingfeng01) [reply paddle/fluid/operators/fused/fused_transformer_op.cc]: updated

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**